v0/destinations/google_cloud_function/transform.js

```diff
@@ -0,0 +1,117 @@
+const {
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  getSuccessRespEvents,
+  getErrorRespEvents
+} = require("../../util");
+
+const {
+  generateBatchedPayloadForArray,
+  generateBatchedPayload,
+  validateDestinationConfig,
+  addHeader
+} = require("./util");
+
+function process(event) {
+  const { message, destination } = event;
+  const { googleCloudFunctionUrl } = destination.Config;
+
+  // Config Validation
+  validateDestinationConfig(destination);
+
+  const response = defaultRequestConfig();
+  // adding header
+  addHeader(response, destination.Config);
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.body.JSON = message;
+  response.endpoint = googleCloudFunctionUrl;
```

Comment:
IMHO we need to add check if https protocol is included in URL, if not then we should add it. WDYT @utsabc ?

Reply:
yes